### PR TITLE
Syntax highlighting for Swift code

### PR DESCRIPTION
**Pull request by yonomitt**
This PR adds support for syntax highlighting for Swift code. There is a new parameter on `Code.init()` called `language`. This is an enum, which can currently be set to `.swift` and `.none` for no syntax highlighting.

This closes issue #8 

I'm not 100% happy with the API, as the language parameter is currently at the end of the parameter list for `Code.init()`, which if you have a large string of code before that, it can be kind of "lost".

Additionally, I think there are two different functions with similar names: `enableHighlight`, which allows you to select single lines of code to go through and syntax highlighting, which is to make colorful code. This may be something that should be looked into.

I'm 100% open to suggestions to either of these two deficiencies (in my view). If no one else thinks they're problematic, I'm happy to go with the group on this one.

https://user-images.githubusercontent.com/8521453/189717086-c5c0d26d-b0b0-4a8f-aa65-14e5438d9373.mov

### CODE REVIEW

File: Sources/DeckUI/DSL/ContentItems/Code.swift [hunk 1]
```diff
@@ -5,32 +5,34 @@
...
 import SwiftUI
 
 public struct Code: ContentItem {
     public let id = UUID()
     let text: String
     let enableHighlight: Bool
+    let language: ProgrammingLanguage
     
-    public init(_ text: String, enableHighlight: Bool = true) {
+    public init(_ text: String, enableHighlight: Bool = true, language: ProgrammingLanguage = .none) {
```
Comment (FullQueueDeveloper) [inline Sources/DeckUI/DSL/ContentItems/Code.swift:17]:
```suggestion
    public init(_ language: ProgrammingLanguage = .none, _ text: String) {
```

and `enableHighlight` could be computed?

```
var enableHighlight: Bool {
  language != .none
}
```

Comment (yonomitt) [reply Sources/DeckUI/DSL/ContentItems/Code.swift:17]: This was another confusing part... the `enableHighlight` allows the user to hit the up and down arrow keys to highlight individual lines of code and go through them while presenting. 

In the video above, it's the black background highlight for the lines (which also makes the font bold and removes syntax highlighting... although those are all configurable)

Comment (FullQueueDeveloper) [reply Sources/DeckUI/DSL/ContentItems/Code.swift:17]: ahhh gotcha

**Comment by FullQueueDeveloper** [comment]
Love it! 💜

File: Sources/DeckUI/DSL/ContentItems/Code.swift (hunk 1 above)
Comment (yonomitt) [reply Sources/DeckUI/DSL/ContentItems/Code.swift]: I've updated to try something similar, but slightly different.... 

```
public init(_ language: ProgrammingLanguage = .none, enableHighlight: Bool = true, text: () -> String)
```

This allows one to do something like:

```
Code(.swift) {
"""
import Splash
import SwiftUI

public struct Code: ContentItem {
    public let id = UUID()
}
"""
}
```

Comment (FullQueueDeveloper) [reply Sources/DeckUI/DSL/ContentItems/Code.swift]: Yes, it's quite elegant 🤩

**Comment by FullQueueDeveloper** [comment]
hmm, @yonomitt I swapped to your branch, but do not see any Syntax highlight? 

```
      Code(.swift) {
        """
        print("hello")
        if 1 < 2 {
          exit(0)
        }
```
Screen Shot 2022-09-13 at 4 39 28 PM

**Comment by yonomitt** [comment]
I think I added a code theme to the default `.dark`, but I didn't touch any of the other themes.

This definitely needs to be documented at some point 😀

**Comment by FullQueueDeveloper** [comment]
@yonomitt `'CodeTheme' initializer is inaccessible due to 'internal' protection level` 🙈

**Comment by FullQueueDeveloper** [comment]
Just presented a talk using this branch and it went great! +1 to merge

**Comment by joshdholtz** [comment]
@FullQueueDeveloper Did you really give a talk with DeckUI?! I would love to link to it from the README if you would like to share it in any form 😊😊😊😊😊😊😊

**Review by joshdholtz — APPROVED** [review]
This is absolutely amazing!! I'm so so so so excited for this 😊 

You da best 🙌

**Comment by yonomitt** [comment]
> @yonomitt `'CodeTheme' initializer is inaccessible due to 'internal' protection level` 🙈
> 
> Could you kindly add a public member-wise initializer?
> 
> ```
>  public init(font: SwiftUI.Font, plainTextColor: SwiftUI.Color, backgroundColor: SwiftUI.Color, tokenColors: [TokenType : SwiftUI.Color]) {
>     self.font = font
>     self.plainTextColor = plainTextColor
>     self.backgroundColor = backgroundColor
>     self.tokenColors = tokenColors
>   }
> ```

Good point! I'll add this shortly

**Comment by FullQueueDeveloper** [comment]
> @FullQueueDeveloper Did you really give a talk with DeckUI?! I would love to link to it from the README if you would like to share it in any form 😊😊😊😊😊😊😊

@joshdholtz yes I did! I'll ping you when the meetup organizers post the recording 😍 And I would be honored for a link in your README ✨

**Comment by FullQueueDeveloper** [comment]
> @FullQueueDeveloper Did you really give a talk with DeckUI?! I would love to link to it from the README if you would like to share it in any form 😊😊😊😊😊😊😊

Yes! The video is up! https://www.youtube.com/watch?v=SxozYjWMhgU 

Also, I published my deck as well! https://github.com/FullQueueDeveloper/Deck-2022-09-13

**Pull request merged**